tests/test_annotations.py

```diff
@@ -11,310 +11,462 @@
+    vtc_segments = am.get_segments(am.annotations[am.annotations["set"] == "vtc_rttm"])
     assert segments.shape[0] == vtc_segments.shape[0]
     assert segments.shape[1] == vtc_segments.shape[1] + 3
 
-    adult_segments = segments[segments['speaker_type'].isin(['FEM', 'MAL'])].sort_values(['segment_onset', 'segment_offset']).reset_index(drop = True)
-    alice = am.get_segments(am.annotations[am.annotations['set'] == 'alice']).sort_values(['segment_onset', 'segment_offset']).reset_index(drop = True)
-    
-    pd.testing.assert_frame_equal(adult_segments[['phonemes', 'syllables', 'words']], alice[['phonemes', 'syllables', 'words']])
+    adult_segments = (
+        segments[segments["speaker_type"].isin(["FEM", "MAL"])]
+        .sort_values(["segment_onset", "segment_offset"])
+        .reset_index(drop=True)
+    )
+    alice = (
+        am.get_segments(am.annotations[am.annotations["set"] == "alice"])
+        .sort_values(["segment_onset", "segment_offset"])
+        .reset_index(drop=True)
+    )
+
+    pd.testing.assert_frame_equal(
+        adult_segments[["phonemes", "syllables", "words"]],
+        alice[["phonemes", "syllables", "words"]],
+    )
+
 
 def test_clipping(project):
     am = AnnotationManager(project)
 
-    input_annotations = pd.read_csv('examples/valid_raw_data/annotations/input.csv')
-    am.import_annotations(input_annotations[input_annotations['set'] == 'vtc_rttm'])
+    input_annotations = pd.read_csv("examples/valid_raw_data/annotations/input.csv")
+    am.import_annotations(input_annotations[input_annotations["set"] == "vtc_rttm"])
     am.read()
 
     start = 1981000
     stop = 1984000
-    segments = am.get_segments(am.annotations[am.annotations['set'] == 'vtc_rttm'])
+    segments = am.get_segments(am.annotations[am.annotations["set"] == "vtc_rttm"])
     segments = am.clip_segments(segments, start, stop)
-    
-    assert segments['segment_onset'].between(start, stop).all() and segments['segment_offset'].between(start, stop).all(), "segments not properly clipped"
-    assert segments.shape[0] == 2, "got {} segments, expected 2".format(segments.shape[0])
+
+    assert (
+        segments["segment_onset"].between(start, stop).all()
+        and segments["segment_offset"].between(start, stop).all()
+    ), "segments not properly clipped"
+    assert segments.shape[0] == 2, "got {} segments, expected 2".format(
+        segments.shape[0]
+    )
+
+
+def test_within_time_range(project):
+    am = AnnotationManager(project)
+    am.project.recordings = pd.read_csv("tests/data/time_range_recordings.csv")
+
+    annotations = pd.read_csv("tests/data/time_range_annotations.csv")
+    matches = am.get_within_time_range(annotations, '09:00', '20:00')
+
+    truth = pd.read_csv('tests/truth/time_range.csv')
+
+    pd.testing.assert_frame_equal(
+        standardize_dataframe(matches, truth.columns),
+        standardize_dataframe(truth, truth.columns)
+    )
+
```

Comment:
maybe i'm missing something, but do you have a check for the user inputting start/stop with wrong formatting? (eg 1pm or 1.13 etc)

Reply:
Added (a7c60cb4c8b14b13221645c89709f96da329c75e)